jobs/gtfs-schedule-validator/gtfs_schedule_validator.py

```diff
@@ -0,0 +1,263 @@
+__version__ = "0.1.0"
+
+import gzip
+import json
+import logging
+import os
+import subprocess
+import tempfile
+import traceback
+from datetime import datetime
+from pathlib import Path
+from typing import Dict, List, ClassVar, Optional
+
+import pendulum
+import typer
+from calitp.storage import (
+    fetch_all_in_partition,
+    GTFSFeedExtractInfo,
+    get_fs,
+    GTFSFeedType,
+    JSONL_GZIP_EXTENSION,
+    PartitionedGCSArtifact,
+    ProcessingOutcome,
+    JSONL_EXTENSION,
+)
+from pydantic import Field, validator
+
+JAVA_EXECUTABLE_PATH_KEY = "GTFS_SCHEDULE_VALIDATOR_JAVA_EXECUTABLE"
+SCHEDULE_VALIDATOR_JAR_LOCATION_ENV_KEY = "GTFS_SCHEDULE_VALIDATOR_JAR"
+JAR_DEFAULT = typer.Option(
+    default=os.environ.get(SCHEDULE_VALIDATOR_JAR_LOCATION_ENV_KEY),
+    help="Path to the GTFS Schedule Validator JAR",
+)
+SCHEDULE_VALIDATION_BUCKET = os.getenv("CALITP_BUCKET__SCHEDULE_VALIDATION")
+
+app = typer.Typer()
+logging.basicConfig()
+
+
+class GTFSScheduleFeedValidation(PartitionedGCSArtifact):
+    bucket: ClassVar[str] = SCHEDULE_VALIDATION_BUCKET
+    table: ClassVar[str] = "validation_reports"
+    partition_names: ClassVar[List[str]] = GTFSFeedExtractInfo.partition_names
+    extract: GTFSFeedExtractInfo = Field(..., exclude={"config"})
+    system_errors: Dict
+
+    @validator("filename", allow_reuse=True)
+    def is_jsonl_gz(cls, v):
+        assert v.endswith(JSONL_GZIP_EXTENSION)
+        return v
+
+    @property
+    def dt(self) -> pendulum.Date:
+        return self.extract.ts.date()
+
+    @property
+    def base64_url(self) -> str:
+        return self.extract.config.base64_encoded_url
+
+    @property
+    def ts(self) -> pendulum.DateTime:
+        return self.extract.ts
+
+
+class GTFSScheduleFeedExtractValidationOutcome(ProcessingOutcome):
+    extract: GTFSFeedExtractInfo = Field(..., exclude={"config"})
+    validation: Optional[GTFSScheduleFeedValidation]
+
+
+# TODO: this and DownloadFeedsResult probably deserve a base class
+class ScheduleValidationResult(PartitionedGCSArtifact):
+    bucket: ClassVar[str] = SCHEDULE_VALIDATION_BUCKET
+    table: ClassVar[str] = "validation_results"
+    partition_names: ClassVar[List[str]] = ["dt"]
```

Comment:
making a note here -- I think we want to use timestamp too? technically if you ran validation once and then ran it again and a download occurred in the interval, you could overwrite with a different file.

Reply:
basically, raises a question about whether `outcomes` is about a validation **action** or about the **latest/current state** of the related artifacts (like the inputs & outputs). if the latter, can keep date-only partition. if the former, should use timestamp

Reply:
I'm going to rename this, it's the file that represents the overall work done by a _job_ so it'll be date-partitioned since that's the Airflow data interval

maybe ScheduleValidationResult?